pypiserver/__main__.py

```diff
@@ -195,17 +195,33 @@ def main(argv=None):
                 err = sys.exc_info()[1]
                 sys.exit("Invalid port(%r) due to: %s" % (v, err))
         elif k in ("-a", "--authenticate"):
-            c.authenticated = [a.lower()
-                               for a in re.split("[, ]+", v.strip(" ,"))
-                               if a]
-            if c.authenticated == ['.']:
-                c.authenticated = []
+            if '{' in v:
+                import ast
+                v = ast.literal_eval(v)
```

Comment:
Given this comes from user input, we may want to try/except `ValueError` and give a useful error message if their input doesn't parse. It's probably also worth doing something like the following to ensure that, if it parsed correctly, that it's the correct format:

```python
try:
  v = ast.literal_eval(v)
except ValueError:
  sys.exit('Could not parse auth string %s! Please ensure string is '
           'correctly formatted.' % v)
if (not isinstance(v, dict)
        or not all([isinstance(i, list) for i in v.values()])):
    sys.exit('Matrix auth string must be a dict of lists. Please see the '
             'README for details.')
````
We could then also roll the next block of logic `if isinstance(v, dict):` into the else for that statement, keeping it all within the if block starting with `if '{' in v:`, since there's no other case in which v will be a dict anyway.